features/steps/header.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from behave import given, then
+
+from docx import Document
+from docx.header import Header
+
+from helpers import test_docx
+
+
+# given ===================================================
+
+@given('a section')
+def given_a_section(context):
+    document = Document(test_docx('doc-default'))
+    context.section = document.sections[0]
```

Comment:
Oh, right, don't know how I missed the doc-default.docx, there it is right there :)

Actually I like your thinking on the single section better than mine there. It's perfectly fine if they don't match exactly, if an existing file does the trick we're not particular :) I'm not sure I'll rebase in a change now that it's merged, but your thinking is right-headed :)

Yes, the first unit test should be on section and be entitled `it_provides_access_to_its_header()`. You'll see examples of that all around if you grep 'it_provides_access_to_' in the tests directory (or ag it as I do :).

You'll want DescribeSection rather than DescribeSections. Sections is the collection of sections in a document, Section is an individual section. It's a little further down in that same file.

The py.test fixtures are pretty cool. They let you separate out the fixture code from the actual test code, which makes things a lot more modular once you get the hang of it.

The other learning curve items in the tests are using fixture params for individual test cases, and mocks (which we'll be using soon). Then I guess the cxml bit, but that's fairly intuitive for most folks it seems, cxml stands for compact XML and it's just a compact notation that's parsed to generate the full text XML snippet.

Let me know where you're at once you've had a quick look around on these and I can provide more.